lib/std/json.zig

```diff
@@ -1905,6 +1910,28 @@ fn parseInternal(
                 else => @compileError("Unable to parse into type '" ++ @typeName(T) ++ "'"),
             }
         },
+        .Vector => |vectorInfo| {
+            switch (token) {
+                .ArrayBegin => {
+                    var vec = std.meta.Vector(vectorInfo.len, vectorInfo.child){};
+                    var idx: usize = 0;
+                    while (true) : (idx += 1) {
+                        const tok = (try tokens.next()) orelse return error.UnexpectedEndOfJson;
+                        switch (tok) {
+                            .ArrayEnd => break,
+                            else => {},
+                        }
+
+                        //we ignore extra data, should this be an option to error or ignore?
```

Comment:
I think it should always error.

Reply:
I added a new error for this in both cases. Smaller and larger than expected vector length.